app/models/request.rb

```diff
@@ -1,4 +1,14 @@
 class Request < ApplicationRecord
   belongs_to :book
   belongs_to :user
+
+  validate :check_valid_date, on: :create
+
+  scope :only_this_user, ->(current_user){where user_id: current_user.id}
```

Comment:
scope này thừa thãi k cần thiết
giả sử có viết thì nên là 
```suggestion
  scope :by_user_ids, ->(user_ids){where user_id: user_ids}
```
để có thể tái sử dụng nhiều lần nhé e